### PR TITLE
Fix Fstat error on test using Debug

### PR DESCRIPTION
**Pull request by bkerler**
This addresses the nasty : 

"AttributeError: '_io.FileIO' object has no attribute 'fstat'"

error when running qiling Test in Debug mode.

### CODE REVIEW

File: qiling/os/posix/syscall/stat.py [hunk 1]
```diff
@@ -213,7 +213,7 @@ def ql_syscall_fstat64(ql, fstat64_fd, fstat64_add, *args, **kw):
 
 def ql_syscall_fstat(ql, fstat_fd, fstat_add, *args, **kw):
 
-    if fstat_fd < 256 and ql.os.fd[fstat_fd] != 0:
+    if fstat_fd < 256 and ql.os.fd[fstat_fd] != 0 and "fstat" in dir(ql.os.fd[fstat_fd]):
```
Comment (cq674350529) [inline qiling/os/posix/syscall/stat.py:216]: maybe use `hasattr(ql.os.fd[fstat_fd], "fstat")` explicitly?

**Comment by bkerler** [comment]
Yeah, very good idea, thanks. Changed it to hasattr :)

**Comment by xwings** [comment]
nice, will wait for the test to complete

**Comment by xwings** [comment]
cool. done!

**Pull request merged**